src/libsyntax/ext/base.rs

```diff
@@ -792,14 +792,16 @@ impl<'a> ExtCtxt<'a> {
     pub fn span_bug(&self, sp: Span, msg: &str) -> ! {
         self.parse_sess.span_diagnostic.span_bug(sp, msg);
     }
-    pub fn trace_macros_diag(&self) {
+    pub fn trace_macros_diag(&mut self) {
         for (sp, notes) in self.expansions.iter() {
             let mut db = self.parse_sess.span_diagnostic.span_note_diag(*sp, "trace_macro");
             for note in notes {
                 db.note(note);
             }
             db.emit();
         }
+        // Fixme: does this result in errors?
+        self.expansions.clear();
```

Comment:
Why is this necessary?

Reply:
As this prevent reporting an expansion trace twice or more